examples/stable-diffusion/training/train_text_to_image_sdxl.py

```diff
@@ -767,7 +776,14 @@ def main(args):
     )
 
     # Load scheduler and models
-    noise_scheduler = DDPMScheduler.from_pretrained(args.pretrained_model_name_or_path, subfolder="scheduler")
+    if args.scheduler == "euler_discrete":
+        noise_scheduler = GaudiEulerDiscreteScheduler.from_pretrained(args.pretrained_model_name_or_path, subfolder="scheduler")
+    elif args.scheduler == "euler_ancestral_discrete":
+        noise_scheduler = GaudiEulerAncestralDiscreteScheduler.from_pretrained(args.pretrained_model_name_or_path, subfolder="scheduler")
+    elif args.scheduler == "ddim":
+        noise_scheduler = GaudiDDIMScheduler.from_pretrained(args.pretrained_model_name_or_path, subfolder="scheduler")
+    else:
+        noise_scheduler = DDPMScheduler.from_pretrained(args.pretrained_model_name_or_path, subfolder="scheduler")
```

Comment:
I think default case can be handled better like this:
```
...
    else:
        noise_scheduler = GaudiDDIMScheduler.from_pretrained(args.pretrained_model_name_or_path, subfolder="scheduler")
        if noise_scheduler.config._class_name == "EulerDiscreteScheduler":
            noise_scheduler = GaudiEulerDiscreteScheduler.from_config(noise_scheduler.config)
        elif noise_scheduler.config._class_name == "EulerAncestralDiscreteScheduler":
            noise_scheduler = GaudiEulerAncestralDiscreteScheduler.from_config(noise_scheduler.config)
```

Reply:
This way we default to proper model original scheduler.  For example, in SD-XL we default to "Euler Discrete" but for Animagine-3.1-XL we default to "Euler Ancestral Discrete"

Reply:
then I'd move first line under last ```else``` so we won't create several schedulers in case if non-default version would be used:
```
...
    else:
        if noise_scheduler.config._class_name == "EulerDiscreteScheduler":
            noise_scheduler = GaudiEulerDiscreteScheduler.from_config(noise_scheduler.config)
        elif noise_scheduler.config._class_name == "EulerAncestralDiscreteScheduler":
            noise_scheduler = GaudiEulerAncestralDiscreteScheduler.from_config(noise_scheduler.config)
        else:
            noise_scheduler = GaudiDDIMScheduler.from_pretrained(args.pretrained_model_name_or_path, subfolder="scheduler")
```

Reply:
Refer to https://huggingface.co/docs/diffusers/v0.8.0/en/using-diffusers/configuration for an example